src/components/TransactionBlock/PlrDaoStakingTransactionBlock.tsx

```diff
@@ -0,0 +1,571 @@
+
+  const theme: Theme = useTheme();
+
+  const fixed = multiCallData?.fixed ?? false;
+  const defaultCustomReceiverAddress =
+    values?.receiverAddress &&
+    !addressesEqual(providerAddress, values?.receiverAddress) &&
+    !addressesEqual(accountAddress, values?.receiverAddress)
+      ? values.receiverAddress
+      : null;
+  const [customReceiverAddress, setCustomReceiverAddress] = useState<string | null>(defaultCustomReceiverAddress);
+  const [useCustomAddress, setUseCustomAddress] = useState<boolean>(!!defaultCustomReceiverAddress);
+
+  const defaultSelectedReceiveAccountType =
+    (!values?.receiverAddress && values?.accountType === AccountTypes.Key) ||
+    (values?.receiverAddress &&
+      values?.accountType === AccountTypes.Contract &&
+      addressesEqual(providerAddress, values?.receiverAddress))
+      ? AccountTypes.Key
+      : AccountTypes.Contract;
+
+  const [selectedReceiveAccountType, setSelectedReceiveAccountType] = useState<string>(
+    defaultSelectedReceiveAccountType
+  );
+
+  const {
+    setTransactionBlockValues,
+    resetTransactionBlockFieldValidationError,
+    setTransactionBlockFieldValidationError,
+  } = useTransactionBuilder();
+
+  useEffect(() => {
+    // Reset transaction block field errors
+    resetTransactionBlockFieldValidationError(transactionBlockId, 'amount');
+    resetTransactionBlockFieldValidationError(transactionBlockId, 'offer');
+    resetTransactionBlockFieldValidationError(transactionBlockId, 'toAssetAddress');
+  }, [selectedFromNetwork]);
+
+  const getWalletBalance = async (chainId: number, name: string) => {
+    return Promise.allSettled(
+      [accountAddress, providerAddress].map(async (address) => {
+        return await getSupportedAssetsWithBalancesForChainId(chainId, true, address);
+      })
+    )
+      .then((response: any[]) => {
+        const accountWalletBalance = response.map(
+          (accountBalance) => accountBalance?.status == 'fulfilled' && accountBalance?.value
+        );
+
+        let smartWalletBalance = 0;
+        let keyBasedBalance = 0;
+
+        accountWalletBalance[0]?.forEach(({ symbol, decimals, balance }) => {
+          if (symbol == plrDaoAsset.symbol) {
+            smartWalletBalance += +ethers.utils.formatUnits(balance, decimals);
+          }
+        });
+        accountWalletBalance[1]?.forEach(({ symbol, decimals, balance }) => {
+          if (symbol == plrDaoAsset.symbol) {
+            keyBasedBalance += +ethers.utils.formatUnits(balance, decimals);
+          }
+        });
+
+        return {
+          chainName: name,
+          keyBasedWallet: keyBasedBalance,
+          smartWallet: smartWalletBalance,
+        };
+      })
+      .catch((error) => {
+        return {
+          chainName: name,
+          keyBasedWallet: 0,
+          smartWallet: 0,
+        };
+      });
+  };
+
+  const getBalanceForAllChains = async () => {
```

Comment:
please rewrite this into:

no then on await
catch with try+catch wrapper
Promise.all to resolve